apps/datasetmanager/api.py

```diff
@@ -64,6 +64,17 @@ def put(self, request, *args, **kwargs):
         self.serializer_class = UpdateDatasetSerializer
         return super(DatasetDetail, self).put(request, args, kwargs)
 
+    def delete(self, request, *args, **kwargs):
+        id = kwargs.get('pk')
+        user = request.user.resource_path
+        dataset = Dataset.objects.get(id=id)
+
+        if dataset.creator_path == user:
```

Comment:
you should also check if request.user.is_god is true so that the admins can also delete other's content